src/app/modules/companies/components/companies-list-filter-bar/companies-list-filter-bar.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, EventEmitter, Output } from '@angular/core';
+import { FilterValues } from './filter-values.model';
+import { InputValues } from './input-values.model';
+
+@Component({
+  selector: 'vc-companies-list-filter-bar',
+  templateUrl: './companies-list-filter-bar.component.html',
+  styleUrls: [
+    './companies-list-filter-bar.component.scss',
+  ],
+})
+export class CompaniesListFilterBarComponent {
+  searchFilterValue = '';
+
+  selectInputValue = '';
+
+  @Output() filterChange = new EventEmitter<FilterValues>();
+
+  onValueChanges({ value, emitterElement } : InputValues): void {
+    switch (emitterElement) {
```

Comment:
Using a different events handler is more preferable.

Reply:
done